src/main/java/net/snowflake/client/core/arrow/VarCharConverter.java

```diff
@@ -143,4 +145,14 @@ public boolean toBoolean(int index) throws SFException {
           SnowflakeUtil.BOOLEAN_STR, str);
     }
   }
+
+  @Override
+  public Date toDate(int index, TimeZone jvmTz, boolean useDateFormat) throws SFException {
+    if (isNull(index)) {
+      return null;
+    } else {
+      throw new SFException(
```

Comment:
why varchar can't convert into date? if value in string is a valid date, we can convert it.

Reply:
That's actually a great point. I got too focused on fixing this for null and the fact that there is a toString for Date type that I didn't consider that there is a valid way to convert a string to a date. Will fix this right now!